### PR TITLE
Add Python 3.13/3.14 support and expand test coverage

### PR DESCRIPTION
**Pull request by shaharbar1**
### Changes:

- Add Python 3.13 and 3.14 to CI/CD matrix in continuous_integration.yml and continuous_delivery.yml
- Replace pd.concat + np.array with np.concatenate in offline_policy_evaluator.py scaler paths — removes unnecessary pandas round-trip
- Add TestFunctionEstimatorEdgeCases in test_offline_policy_evaluator.py — covers validation errors, predict-before-fit, and multiclass probability extraction
- Add TestValidateLoggedData — parametrized error-path tests for _validate_logged_data
- Add TestOfflinePolicyEvaluatorPipeline — covers dict-scaler extraction, estimate_policy with n_cores=0, and update_and_evaluate with with_test=True
- Add TestMabPredictSerializedUtils — covers _mab_predict_serialized error paths and verbose logging
- Add strategy edge-case tests in test_strategy.py — covers refine_p empty/rejected inputs, ClassicBandit optimization failure, _solve_nc_subproblem feasibility branches, and _find_utopia_reference_point zero-denominator/LinAlgError fallbacks
- Update authors

## Summary by CodeRabbit

* **New Features**
  * Added support for Python 3.13 and 3.14.

* **Dependencies**
  * Expanded supported Python range to <3.15; refined NumPy/SciPy constraints for newer Pythons; removed tox dev dependency.

* **Documentation**
  * Updated authors list with additional contributors.

* **Chores**
  * Package version bumped to 7.1.0; CI/CD workflows updated to test/publish on Python 3.13–3.14; removed tox.ini.

* **Tests**
  * Expanded test coverage with many new edge-case and pipeline tests.

[Review Change Stack](https://app.coderabbit.ai/change-stack/PlaytikaOSS/pybandits/pull/144?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -20,22 +22,24 @@ classifiers = [
...
+    "Programming Language :: Python :: 3.13",
+    "Programming Language :: Python :: 3.14",
 ]
 homepage = "https://github.com/PlaytikaOSS/pybandits"
 repository = "https://github.com/PlaytikaOSS/pybandits"
 keywords = ["multi-armed bandits", "reinforcement-learning", "optimization"]
 
 [tool.poetry.dependencies]
-python = ">=3.9,<3.13"
+python = ">=3.9,<3.15"
```
Comment (AnastasiiaKabeshova) [inline pyproject.toml:33]: Please update poetry.lock as well, because I think poetry install fail in current state

File: pyproject.toml [hunk 2]
```diff
@@ -20,22 +22,24 @@ classifiers = [
...
 numpy = [
     { version = ">=1.25", python = ">=3.9,<3.12" },
-    { version = ">=1.26", python = "3.12.*" },
+    { version = ">=1.26", python = ">=3.12" },
 ]
 pydantic = "^2.0"
 scipy = [
     { version = "^1.9", python = ">=3.9,<3.12" },
-    { version = "^1.11", python = "3.12.*" },
+    { version = "^1.11", python = ">=3.12" },
```
Comment (AnastasiiaKabeshova) [inline pyproject.toml]: Claude suggested:
pyproject.toml keeps scipy = {version = "^1.11", python = ">=3.12"}. scipy 3.13 support only arrived in 1.14.0 — ^1.11 has no 3.13 wheels.

**Review by AnastasiiaKabeshova — APPROVED** [review]

File: pyproject.toml (hunk 1 above)
Comment (shaharbar1) [reply pyproject.toml:33]: We don't commit the lock file
